src/tests/libpdbg_release_dt_root_test.c

```diff
@@ -0,0 +1,358 @@
+/* Copyright 2023 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <inttypes.h>
+#include <assert.h>
+
+#include <libpdbg.h>
+
+static int count_target(struct pdbg_target *parent, const char *classname)
+{
+    struct pdbg_target *target = NULL;
+    int cnt = 0;
+
+    pdbg_for_each_target(classname, parent, target)
+        ++cnt;
+
+    return cnt;
+}
+
+static int count_class_target(const char *classname)
```

Comment:
this file should just have init device tree with fake DB, get the count of targets, now clear the targets and then if possible check the count again like using count_class_target, or else you can define a method where it gives count of targets or classes and you can validate here afte clearing the targets

Reply:
Changed as suggested